scripts/generate.js

```diff
@@ -27,8 +43,38 @@ const ${name}Icon = ({ width = 24, height = 24, viewBox = '0 0 24 24', className
   );
 };
 
-export default ${name}Icon;
+export default ${nameIcon};
+`;
+  fs.writeFileSync(Path.join(buildPath, `${nameIcon}.js`), fileContent);
+
+  // create the [name].d.ts contents
+  const typeContent =
+`import { ComponentType } from "react";
+import { ReactMdiIconProps } from './typings'
+
+declare const ${nameIcon}: React.ComponentType<ReactMdiIconProps>;
+export default ${nameIcon};
 `;
+  fs.writeFileSync(Path.join(publishPath, `${nameIcon}.d.ts`), typeContent);
+}
+
+// create the global typings.d.ts
+const typingsContent =
+`import * as React from "react";
 
-  fs.writeFileSync(`${__dirname}/../build/${name}Icon.js`, fileContent);
+export interface ReactMdiIconProps {
+  width?: number
+  height?: number
+  viewBox?: string
+  className?: string
 }
+
+${icons.map(nameIcon =>
+`declare const ${nameIcon}: React.ComponentType<ReactMdiIconProps>;
+`).join('')}
+`;
+fs.writeFileSync(Path.join(publishPath, 'typings.d.ts'), typingsContent);
+
+// Build the index.js file, before babel compile
+const indexContent = icons.map(nameIcon => `export ${nameIcon} from './${nameIcon}'`).join('\n')
+fs.writeFileSync(Path.join(buildPath, 'index.js'), indexContent);
```

Comment:
Personally I'm against having this `index.js`.
It was already there and I removed it for a reason. It's just too easy to include all icons in your browser bundle this way by accident.

Reply:
I know what you mean, before there was webpack I'd said the same thing. But now we do have dead code elimination and tree shaking almost as the default.
There are even rules to support this: https://palantir.github.io/tslint/rules/no-submodule-imports/

The rule is actually the reason I did include it...

Libraries like lodash do this as well. If you want I'll remove it, but I would like it in there.

Reply:
Let's merge it in without the `index.js` for now. I will investigate again if it can be added back without issues 🙂

Reply:
Yeah, the last version does not have the index.js (and typings for it) in it. I pushed this about 2hours ago